src/QueryBuilder/QueryBuilder/helpers/query.js

```diff
@@ -173,14 +173,14 @@ const getFormattedSourceField = async ({ item, intl, booleanOptions, fieldOption
     const hasSourceOrValues = values || source;
     let formattedValue;
 
-    if (Array.isArray(value) && source) {
-      const params = await getParamsSource?.({
+    if (Array.isArray(value) && hasSourceOrValues) {
+      const params = source ? await getParamsSource?.({
         entityTypeId: source?.entityTypeId,
         columnName: source?.columnName,
         searchValue: '',
-      });
+      }).then((data) => data?.content) : values;
```

Comment:
It is difficult to follow this ternary, especially with the `:` in the middle of a line several lines after its corresponding `?`. Instead of 
```
condition ? if-stuff
  more if-stuff
  still more if-stuff : else-stuff;
```
 Maybe structure it like 
```
condition ?
  if-stuff
  :
  else-stuff
```
or just assign the default and then reassign like
```
const params = values;
if (source) { params = ... }
```

Reply:
Hi Zak, I initially didn’t like the option with the ternary operator either, I tried to format it now, but it seems that in this case a simple additional “if” looks more readable. Thanks for pointing it